packages/devtools_app/lib/src/code_size/code_size_processor.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../charts/treemap.dart';
+import 'canned_data.dart';
+import 'code_size_controller.dart';
+
+class CodeSizeProcessor {
+  CodeSizeProcessor(this.codeSizeController);
+
+  final CodeSizeController codeSizeController;
+
+  void loadJson() {
+    final TreemapNode root = buildTreeFromJson();
+    codeSizeController.changeRoot(root);
+  }
+
+  TreemapNode addChildNode(TreemapNode parent, String name, int byteSize) {
+    TreemapNode child = parent.childrenMap[name];
+    if (child == null) {
+      child = TreemapNode(
+        name: name,
+        byteSize: byteSize,
+        childrenMap: <String, TreemapNode>{},
+      );
+      parent.childrenMap[child.name] = child;
+      parent.addChild(child);
+    } else {
+      child.byteSize += byteSize;
+    }
+    return child;
+  }
+
+  TreemapNode buildTreeFromJson() {
+    final List data = jsonDecode(sizesJson);
+
+    final rootChildren = <String, TreemapNode>{};
+
+    final TreemapNode root = TreemapNode(
+      name: 'Root',
+      childrenMap: rootChildren,
+    );
+
+    // Can optimize look up / retrieve time with a hashmap
+    for (dynamic memoryUsage in data) {
+      String libraryName = memoryUsage['l'];
+      if (libraryName == null || libraryName == '') {
+        libraryName = 'Unnamed Library';
+      }
+      String className = memoryUsage['c'];
+      if (className == null || className == '') {
+        className = 'Unnamed Class';
+      }
+      String methodName = memoryUsage['n'];
+      if (methodName == null || methodName == '') {
+        methodName = 'Unnamed Method';
+      }
+      final int byteSize = memoryUsage['s'];
+      if (byteSize == null) {
+        throw 'Size was null for $memoryUsage';
+      }
+      root.byteSize += byteSize;
+
+      TreemapNode libraryLevelChild;
+      if (libraryName.startsWith('package:flutter/src/')) {
```

Comment:
why is `package:flutter` handled differently? it would be nice if we handled all large packages consistently. There isn't anything that special about how the files are laid out under package:flutter.

Reply:
lets figure out a suitably complex test application. if your application uses multiple third party packages you will see this sort of pattern in places outside package:flutter.